velox/type/TimestampConversion.cpp

```diff
@@ -579,26 +579,24 @@ int64_t fromDateString(const char* str, size_t len) {
   return daysSinceEpoch;
 }
 
-int32_t
-castFromDateString(const char* str, size_t len, bool isNonStandardCast) {
+int32_t castFromDateString(const char* str, size_t len, bool isStandardCast) {
   int64_t daysSinceEpoch;
   size_t pos = 0;
 
-  auto mode = isNonStandardCast ? ParseMode::kNonStandardCast
-                                : ParseMode::kStandardCast;
+  auto mode =
+      isStandardCast ? ParseMode::kStandardCast : ParseMode::kNonStandardCast;
   if (!tryParseDateString(str, len, pos, daysSinceEpoch, mode)) {
-    if (isNonStandardCast) {
+    if (isStandardCast) {
       VELOX_USER_FAIL(
           "Unable to parse date value: \"{}\"."
-          "Valid date string patterns include "
-          "(YYYY, YYYY-MM, YYYY-MM-DD), and any pattern prefixed with [+-]",
+          "Valid date string pattern is (YYYY-MM-DD), "
```

Comment:
Exceptions are expensive and cause performance issues when queries use try_cast and have many failing rows. It would be nice to re-write this logic to not use exceptions.